### ISSUE TITLE
Unable to rename and/or delete files/folders in a GroupFolder

### ISSUE THREAD

**Issue by JOduMonT** [nextcloud/groupfolders#1181]
Same here, I'm unable to rename any GroupFolder

- Nextcloud version 20
- Nextcloud Group Folder 8.2.0

I used FlowUpload to upload few GiB in 5 differents GroupFolders

When I try to rename the content inside these GroupFolders
- via WebDav I simply receive an `Error: Internal Server Error`
- via the WebUI the message is: `Could not rename "GroupFolderName"✖`
I don't have a SSH access on this instance.

### now I'm stuck, because I can't even move and/or delete the data of these GroupFolders

_Originally posted by @JOduMonT in https://github.com/nextcloud/groupfolders/issues/290#issuecomment-745723036_

**Comment by Lars1963** [comment]
I'm expiring the same issue

**Comment by JOduMonT** [comment]
> I'm expiring the same issue

I guess the only thing we could do, is deleting these directories with the console than re-indexing the files with occ
https://docs.nextcloud.com/server/15/admin_manual/configuration_server/occ_command.html#file-operations-label

**Comment by chridiby** [comment]
Hi, i am new to NextCloud and doing some tests.
I'm expiring the same issue.
Can you try something?
On my installation, i have "advanced permissions" activated on the group folders - settings.
After deactivating "advanced permissions", i am able to rename and move the folders.
Activating them again -> no renaming is possible.

Maybe that helps...

**Comment by JOduMonT** [comment]
>  Activating them again -> no renaming is possible.

on my side advanced permissions is not enabled

**Comment by pierreozoux** [comment]
In my case, I found that bug too.

I could delete but couldn't move nor rename.

The issue is a `é` in the username of the user.

This looks like related to https://github.com/nextcloud/server/issues/18298

and got this response in the browser logs:

```
<?xml version="1.0" encoding="utf-8"?>
<d:error xmlns:d="DAV:" xmlns:s="http://sabredav.org/ns">
  <s:exception>Sabre\DAV\Exception\Forbidden</s:exception>
  <s:message>Requested uri () is out of base uri (/remote.php/dav/)</s:message>
</d:error>
```

This was the requested path:
```
MOVE | https://cloud.tld.org/remote.php/dav/files/h%C3%A9l%C3%A8ne/groupfolder/test-deplacer/test1
```

Username is `hélène` that should be encoded: `h%C3%A9l%C3%A8ne` which is correct.

I'm wondering if it is nginx or php or nextcloud that is the responsible for this error, I'll continue digging.

**Comment by Daksol** [comment]
From my use of Group Folders I have found that it is never possible to rename them - except possibly before they have any content. And you cannot rename Groups either, so it pays to get all that detail right first time.

Not sure if I understand the problem being encountered by @Jodumont, but these steps may be useful

Standard GUI for the Group Folders configuration allows anyone in the Admin group to update the permissions for any given folder, with a URL like
__https://SOMEHOSTNAME/index.php/settings/admin/groupfolders

* Suppose you are a member of Group ABC
* Then an Admin Group member can grant Write+Update+Share to Group ABC for any folders of interest
* That will allow you to move the relevant files out to a location where you have read/write
* Once new Group Folders have been created move the files back in.
* Delete original incorrectly named GFs.

**Comment by pierreozoux** [comment]
@JOduMonT was there special char in the username that got this error?

**Comment by JOduMonT** [comment]
> @JOduMonT was there special char in the username that got this error?

not in the username nor in a directory name, but probably in a files

**Comment by vdietmar** [comment]
I have just run into the same issue, i.e. I cannot rename a folder that I created in a Group Folder. The user who created the folder has a special (German) character in his clear name (nor user name). The folder does not have any special characters. Other folders in the same Group Folder can be renamed.

I am stuck as I don't have access to OCC (hosted/managed Nextcloud at IONOS).

Is this still open? Not sure I agree with it being qualified as "No or low priority", currently.

Nextcloud: 24.0.7 Enterprise
Group Folders: 12.0.3

**Comment by mkbrechtel** [comment]
@fschrempf I can confirm that this problem is there in Nextcloud 25.0.6.1. I will try to update our instalation and check if it also occurs in Nextcloud 26

**Comment by fschrempf** [comment]
> @fschrempf Please reevalute the priority of this ticket.

Actually I'm just a community member who helped out with issue triaging in the past. The assigned priority doesn't have a meaning at all as the chances are low that any "active" NC developer even cares at all.

**Comment by joshtrichards** [comment]
@JOduMonT wrote:

> via WebDav I simply receive an Error: Internal Server Error

Can you (or anyone really) provide the stack trace associated with one of these transactions from your `nextcloud.log`?

**Comment by Fiodin** [comment]
I am having the same problem and i can't delete folders inside a groupfolder.

Here is my log when trying to delete a folder:
[webdav] Fehler: TypeError: OC\Files\Cache\QuerySearchHelper::getCachesAndMountPointsForSearch(): Argument #1 ($root) must be of type OC\Files\Node\Root, OC\Files\Node\LazyRoot given, called in /var/www/cloud/lib/private/Files/Node/Folder.php on line 238 at <<closure>>

 0. /var/www/cloud/lib/private/Files/Node/Folder.php line 238
    OC\Files\Cache\QuerySearchHelper->getCachesAndMountPointsForSearch()
 1. /var/www/cloud/lib/private/Files/Node/Folder.php line 292
    OC\Files\Node\Folder->search()
 2. /var/www/cloud/apps/music/lib/Utility/Scanner.php line 380
    OC\Files\Node\Folder->searchByMime()
 3. /var/www/cloud/apps/music/lib/Hooks/FileHooks.php line 42
    OCA\Music\Utility\Scanner->deleteFolder()
 4. <<closure>>
    OCA\Music\Hooks\FileHooks::deleted()
 5. /var/www/cloud/lib/private/Hooks/EmitterTrait.php line 105
    call_user_func_array()
 6. /var/www/cloud/lib/private/Hooks/PublicEmitter.php line 40
    OC\Hooks\BasicEmitter->emit()
 7. /var/www/cloud/lib/private/Files/Node/Root.php line 143
    OC\Hooks\PublicEmitter->emit()
 8. <<closure>>
    OC\Files\Node\Root->emit()
 9. /var/www/cloud/lib/private/Files/Node/LazyFolder.php line 73
    call_user_func_array()
10. /var/www/cloud/lib/private/Files/Node/LazyFolder.php line 101
    OC\Files\Node\LazyFolder->__call()
11. /var/www/cloud/lib/private/Files/Node/HookConnector.php line 145
    OC\Files\Node\LazyFolder->emit()
12. /var/www/cloud/lib/private/legacy/OC_Hook.php line 105
    OC\Files\Node\HookConnector->delete()
13. /var/www/cloud/lib/private/Files/View.php line 1257
    OC_Hook::emit()
14. /var/www/cloud/lib/private/Files/View.php line 1126
    OC\Files\View->runHooks()
15. /var/www/cloud/lib/private/Files/View.php line 321
    OC\Files\View->basicOperation()
16. /var/www/cloud/apps/dav/lib/Connector/Sabre/Directory.php line 307
    OC\Files\View->rmdir()
17. /var/www/cloud/3rdparty/sabre/dav/lib/DAV/Tree.php line 179
    OCA\DAV\Connector\Sabre\Directory->delete()
18. /var/www/cloud/3rdparty/sabre/dav/lib/DAV/CorePlugin.php line 281
    Sabre\DAV\Tree->delete()
19. /var/www/cloud/3rdparty/sabre/event/lib/WildcardEmitterTrait.php line 89
    Sabre\DAV\CorePlugin->httpDelete()
20. /var/www/cloud/3rdparty/sabre/dav/lib/DAV/Server.php line 472
    Sabre\DAV\Server->emit()
21. /var/www/cloud/3rdparty/sabre/dav/lib/DAV/Server.php line 253
    Sabre\DAV\Server->invokeMethod()
22. /var/www/cloud/3rdparty/sabre/dav/lib/DAV/Server.php line 321
    Sabre\DAV\Server->start()
23. /var/www/cloud/apps/dav/lib/Server.php line 364
    Sabre\DAV\Server->exec()
24. /var/www/cloud/apps/dav/appinfo/v2/remote.php line 35
    OCA\DAV\Server->exec()
25. /var/www/cloud/remote.php line 172
    require_once("/var/www/cloud/ ... p")

DELETE /remote.php/dav/files/popelhans/Fotos/test
from 91.47.254.71 by popelhans at 2023-10-17T09:14:06+00:00

**Comment by joshtrichards** [comment]
Duplicate of owncloud/music#1068
Fixed in nextcloud/server#38625
